auth/auth/auth.py

```diff
@@ -136,6 +137,74 @@ def get_flow(redirect_uri, state=None):
     })
 
 
+@routes.get('/users')
+@web_authenticated_developers_only()
+async def get_users(request, userdata):
+    dbpool = request.app['dbpool']
+    async with dbpool.acquire() as conn:
+        async with conn.cursor() as cursor:
+            await cursor.execute('SELECT * FROM users;')
+            users = await cursor.fetchall()
+    page_context = {
+        'users': users
+    }
+    return await render_template('auth', request, userdata, 'users.html', page_context)
+
+
+@routes.post('/users')
+@check_csrf_token
+@web_authenticated_developers_only()
+async def post_create_user(request, userdata):  # pylint: disable=unused-argument
+    session = await aiohttp_session.get_session(request)
+    dbpool = request.app['dbpool']
+    post = await request.post()
+    username = post['username']
+    email = post['email']
+    is_developer = post.get('is_developer') == '1'
+
+    async with dbpool.acquire() as conn:
+        async with conn.cursor() as cursor:
+            await cursor.execute(
+                '''
+INSERT INTO users (state, username, email, is_developer)
+VALUES (%s, %s, %s, %s);
+''',
+                ('creating', username, email, is_developer))
+            user_id = cursor.lastrowid
+
+    set_message(session, f'Created user {user_id} {username}.', 'info')
+
+    return web.HTTPFound(deploy_config.external_url('auth', '/users'))
+
+
+@routes.post('/users/delete')
+@check_csrf_token
+@web_authenticated_developers_only()
+async def delete_user(request, userdata):  # pylint: disable=unused-argument
+    session = await aiohttp_session.get_session(request)
+    dbpool = request.app['dbpool']
+    post = await request.post()
+    id = post['id']
+    username = post['username']
+
+    async with dbpool.acquire() as conn:
+        async with conn.cursor() as cursor:
+            n_rows = await cursor.execute(
+                '''
+UPDATE users
+SET state = 'deleting'
+WHERE id = %s AND username = %s;
+''',
+                (id, username))
+            if n_rows != 1:
+                assert n_rows == 0
+                set_message(session, f'Delete failed, no such user {id} {username}.', 'error')
+
+    set_message(session, f'Deleted user {id} {username}.', 'info')
```

Comment:
Similarly I find the semantics a bit confusing: if database state is deleting could you help me understand why we return “deleted”

Reply:
This is going to be a common pattern: the deletion process is a long-lived process involving multiple service, so we mark it as being deleted, it needs to be treated as deleted by the rest of the application (e.g. a deleting user can no longer log in) while the deletion process finishes.  So from the perspective of the user this message is literally correct.

Reply:
Sounds good. Interested at some point in working on the service, so will understand more about it then.